server/routes/appointments/create-and-edit/handlers/bulk-appointments/reviewBulkAppointment.ts

```diff
@@ -0,0 +1,95 @@
+import { Transform, plainToInstance } from 'class-transformer'
+import { Request, Response } from 'express'
+import { ValidateNested } from 'class-validator'
+import SimpleTime from '../../../../../commonValidationTypes/simpleTime'
+
+// class-validator will not validate against object unknown properties, but it will validate a map.
+// This hack transforms the object to a map for validation
+// This seems to be a limitation of class-validator, https://github.com/typestack/class-validator/issues/1011
+export class AppointmentTimes {
+  @Transform(({ value }) =>
+    Object.keys(value).reduce((acc, k) => acc.set(k, plainToInstance(SimpleTime, value[k])), new Map()),
+  )
+  @ValidateNested()
+  startTime: Map<string, SimpleTime>
```

Comment:
Maybe rename these to startTime**s**/endTime**s** so it's clear it's a collection?